Classes/Controller/AbstractController.php

```diff
@@ -88,9 +87,28 @@ public function injectDocumentRepository(DocumentRepository $documentRepository)
     protected function initialize()
     {
         $this->requestData = GeneralUtility::_GPmerged('tx_dlf');
-        if (empty($this->requestData['page'])) {
+
+        // Sanitize user input to prevent XSS attacks.
+
+        // tx_dlf[id] may only be an UID or URI.
+        if (
+            !empty($this->requestData['id'])
+            && !MathUtility::canBeInterpretedAsInteger($this->requestData['id'])
+            && !GeneralUtility::isValidUrl($this->requestData['id'])
+        ) {
+            unset($this->requestData['id']);
+            $this->logger->error('Invalid ID or URI "' . $this->requestData['id'] . '" for document loading');
+        }
+
+        // tx_dlf[page] may only be a positive integer or valid XML ID.
+        if (
+            empty($this->requestData['page'])
+            || !preg_match('/^[1-9][0-9]*$|^[_a-z][_a-z0-9-.]*$/', $this->requestData['page']) === 1
```

Comment:
Why do you use the MathUtility::canBeInterpretedAsInteger above and here is a regex for the integer check?

Reply:
Because I tried to incorporate both checks (for integer and valid XML IDs) into one statement. Otherwise I'd have to nest multiple `||` and `&&` operations into this condition check, which would be even more confusing, I guess.

Reply:
So maybe move the checks to own method to avoid long conditions in if statement?

Reply:
You are right. I hope it's easier to read now.

Reply:
Actually maybe it would be good just put the sanitation of the input in the own function:

private function sanitizeUserInput() {
    // Sanitize user input to prevent XSS attacks.

        // tx_dlf[id] may only be an UID or URI.
        if (
            !empty($this->requestData['id'])
            && !MathUtility::canBeInterpretedAsInteger($this->requestData['id'])
            && !GeneralUtility::isValidUrl($this->requestData['id'])
        ) {
            $this->logger->error('Invalid ID or URI "' . $this->requestData['id'] . '" for document loading');
            unset($this->requestData['id']);
        }

        // tx_dlf[page] may only be a positive integer or valid XML ID.
        if (
            !empty($this->requestData['page'])
            && !MathUtility::canBeInterpretedAsInteger($this->requestData['page'])
            && !Helper::isValidXmlId($this->requestData['page'])
        ) {
            $this->requestData['page'] = 1;
        }
}

Also logger for $this->requestData['id'] needs to be called before unsetting the value.

Reply:
> Also logger for $this->requestData['id'] needs to be called before unsetting the value.

Uh, this is embarrassing... ;o)

Reply:
OK, I moved the input sanitation to a separate method.

Reply:
Great! The code is ready merge :)